src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1127,19 +1128,31 @@ def decrypt_key(cmd, client, algorithm, value, iv=None, tag=None, aad=None,
 
 
 def sign_key(cmd, client, algorithm, digest, name=None, version=None):
+    if '256' in algorithm:
+        hash_func = hashlib.sha256
+    elif '384' in algorithm:
+        hash_func = hashlib.sha384
+    else:
+        hash_func = hashlib.sha512
     SignatureAlgorithm = cmd.loader.get_sdk('SignatureAlgorithm', mod='crypto._enums',
                                             resource_type=ResourceType.DATA_KEYVAULT_KEYS)
     crypto_client = client.get_cryptography_client(name, key_version=version)
-    return crypto_client.sign(SignatureAlgorithm(algorithm), digest.encode('utf-8'))
+    return crypto_client.sign(SignatureAlgorithm(algorithm),
+                              hash_func(base64.b64decode(digest.encode('utf-8'))).digest())
 
 
 def verify_key(cmd, client, algorithm, digest, signature, name=None, version=None):
-    import base64
+    if '256' in algorithm:
+        hash_func = hashlib.sha256
+    elif '384' in algorithm:
+        hash_func = hashlib.sha384
+    else:
+        hash_func = hashlib.sha512
     SignatureAlgorithm = cmd.loader.get_sdk('SignatureAlgorithm', mod='crypto._enums',
                                             resource_type=ResourceType.DATA_KEYVAULT_KEYS)
     crypto_client = client.get_cryptography_client(name, key_version=version)
     return crypto_client.verify(SignatureAlgorithm(algorithm),
-                                digest.encode('utf-8'),
+                                hash_func(base64.b64decode(digest.encode('utf-8'))).digest(),
```

Comment:
The extra round of digest is going to be an unwelcome surprise for users, who won't expect it, if they try to verify using their own code. There needs to be a testcase that checks that signatures created using the REST API directly can be verified with this call.

Reply:
I agree that CLI shouldn't make extra undocumented logic above REST API. will revert additional hash